storage/reads/array_cursor.go

```diff
@@ -18,84 +17,52 @@ func (v *singleValue) Value(key string) (interface{}, bool) {
 }
 
 func newAggregateArrayCursor(ctx context.Context, agg *datatypes.Aggregate, cursor cursors.Cursor) cursors.Cursor {
```

Comment:
Since you have the `Agg` section in your template file now, I feel this part of the code should also be generated.